spec/requests/api/roles_spec.rb

```diff
@@ -0,0 +1,375 @@
+      role = FactoryGirl.create(:miq_user_role_miq_request_approver)
+
+      new_feature = {:identifier => "miq_request_view"}
+      url = "#{roles_url}/#{role.id}/features"
+      run_post(url, gen_request(:assign, new_feature))
+
+      expect_request_success
+      expect_result_resources_to_include_keys("results", %w(id name read_only))
+
+      # Refresh the role object
+      role = MiqUserRole.find(role.id)
+
+      # Confirm original feature
+      expect(role.allows?(:identifier => 'miq_request_approval')).to be_true
+
+      # Confirm new feature
+      expect(role.allows?(new_feature)).to be_true
+    end
+
+    it "supports assigning multiple product features" do
+      api_basic_authorize collection_action_identifier(:roles, :edit)
+      role = FactoryGirl.create(:miq_user_role_miq_request_approver)
+
+      url = "#{roles_url}/#{role.id}/features"
+      run_post(url, gen_request(:assign, features_list))
+
+      expect_request_success
+      expect_result_resources_to_include_keys("results", %w(id name read_only))
+
+      # Refresh the role object
+      role = MiqUserRole.find(role.id)
+
+      # Confirm original feature
+      expect(role.allows?(:identifier => 'miq_request_approval')).to be_true
+
+      # Confirm new features
+      features_list['features'].each do |feature|
+        expect(role.allows?(feature)).to be_true
+      end
+    end
+
+    it "supports un-assigning just a single product feature" do
+      api_basic_authorize collection_action_identifier(:roles, :edit)
+      role = FactoryGirl.create(:miq_user_role, :miq_product_features => @product_features)
+
+      removed_feature = {:identifier => "ems_infra_tag"}
+      url = "#{roles_url}/#{role.id}/features"
+      run_post(url, gen_request(:unassign, removed_feature))
+
+      expect_request_success
+      expect_result_resources_to_include_keys("results", %w(id name read_only))
+
+      # Refresh the role object
+      role = MiqUserRole.find(role.id)
+
+      @product_features.each do |feature|
+        expect(role.allows?(feature)).to be_true unless feature[:identifier].eql?('ems_infra_tag')
+        expect(role.allows?(feature)).to be_false if feature[:identifier].eql?('ems_infra_tag')
+      end
+    end
+
+    it "supports un-assigning multiple product features" do
+      api_basic_authorize collection_action_identifier(:roles, :edit)
+      role = FactoryGirl.create(:miq_user_role, :miq_product_features => @product_features)
+
+      url = "#{roles_url}/#{role.id}/features"
+      run_post(url, gen_request(:unassign, features_list))
+
+      expect_request_success
+      expect_result_resources_to_include_keys("results", %w(id name read_only))
+
+      # Refresh the role object
+      role = MiqUserRole.find(role.id)
+
+      # Confirm requested features removed first, and others remain
+      @product_features.each do |feature|
+        expect(role.allows?(feature)).to be_true unless features_list['features'].find do |removed_feature|
```

Comment:
do you think these tests could benefit from being DRY'd up a bit? also wondering, it maybe more idiomatic (and limit upgrade pains) to use RSpec's predicate matchers here and elsewhere. maybe something like:

```ruby
if in_features_list?(feature)
  expect(role).not_to allow(feature)
else
  expect(role).to allow(feature)
end
```